lib/msf/core/payload/stager.rb

```diff
@@ -188,7 +188,12 @@ def handle_connection(conn, opts={})
         end
       end
 
-      p = generate_stage(opts)
+      begin
+        p = generate_stage(opts)
+      rescue ::MetasploitPayloads::NotFoundError, ::MetasploitPayloads::NotReadableError => e
+        print_error(e.message)
+        raise e
+      end
```

Comment:
We can move this down into the `encode_stage` scenario I think


```
rescue ::RuntimeError => e
        warning_msg = "Failed to stage"
        warning_msg << " (#{conn.peerhost})"  if conn.respond_to? :peerhost
        warning_msg << ": #{e}"
```

Note that I swapped `$!` to `e` here as well